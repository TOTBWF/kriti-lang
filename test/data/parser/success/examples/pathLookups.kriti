# Path Lookups
[ {{ $foo }},
  {{ $foo.bar.baz }},
  {{ $foo['bar baz'].qux }},
  {{ $foo[1][0][3].ok }}
]