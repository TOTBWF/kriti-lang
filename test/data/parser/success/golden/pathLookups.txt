Array
    ( Span
        { start = AlexSourcePos
            { line = 1
            , col = 1
            }
        , end = AlexSourcePos
            { line = 5
            , col = 2
            }
        }
    )
    [ Path
        ( Span
            { start = AlexSourcePos
                { line = 1
                , col = 3
                }
            , end = AlexSourcePos
                { line = 1
                , col = 13
                }
            }
        )
        [ Obj
            ( Span
                { start = AlexSourcePos
                    { line = 1
                    , col = 6
                    }
                , end = AlexSourcePos
                    { line = 1
                    , col = 10
                    }
                }
            ) "$foo"
        ]
    , Path
        ( Span
            { start = AlexSourcePos
                { line = 2
                , col = 3
                }
            , end = AlexSourcePos
                { line = 2
                , col = 21
                }
            }
        )
        [ Obj
            ( Span
                { start = AlexSourcePos
                    { line = 2
                    , col = 6
                    }
                , end = AlexSourcePos
                    { line = 2
                    , col = 10
                    }
                }
            ) "$foo"
        , Obj
            ( Span
                { start = AlexSourcePos
                    { line = 2
                    , col = 10
                    }
                , end = AlexSourcePos
                    { line = 2
                    , col = 14
                    }
                }
            ) "bar"
        , Obj
            ( Span
                { start = AlexSourcePos
                    { line = 2
                    , col = 14
                    }
                , end = AlexSourcePos
                    { line = 2
                    , col = 18
                    }
                }
            ) "baz"
        ]
    , Path
        ( Span
            { start = AlexSourcePos
                { line = 3
                , col = 3
                }
            , end = AlexSourcePos
                { line = 3
                , col = 28
                }
            }
        )
        [ Obj
            ( Span
                { start = AlexSourcePos
                    { line = 3
                    , col = 6
                    }
                , end = AlexSourcePos
                    { line = 3
                    , col = 10
                    }
                }
            ) "$foo"
        , Obj
            ( Span
                { start = AlexSourcePos
                    { line = 3
                    , col = 10
                    }
                , end = AlexSourcePos
                    { line = 3
                    , col = 21
                    }
                }
            ) "bar baz"
        , Obj
            ( Span
                { start = AlexSourcePos
                    { line = 3
                    , col = 21
                    }
                , end = AlexSourcePos
                    { line = 3
                    , col = 25
                    }
                }
            ) "qux"
        ]
    , Path
        ( Span
            { start = AlexSourcePos
                { line = 4
                , col = 3
                }
            , end = AlexSourcePos
                { line = 4
                , col = 25
                }
            }
        )
        [ Obj
            ( Span
                { start = AlexSourcePos
                    { line = 4
                    , col = 6
                    }
                , end = AlexSourcePos
                    { line = 4
                    , col = 10
                    }
                }
            ) "$foo"
        , Arr
            ( Span
                { start = AlexSourcePos
                    { line = 4
                    , col = 10
                    }
                , end = AlexSourcePos
                    { line = 4
                    , col = 13
                    }
                }
            ) 1
        , Arr
            ( Span
                { start = AlexSourcePos
                    { line = 4
                    , col = 13
                    }
                , end = AlexSourcePos
                    { line = 4
                    , col = 16
                    }
                }
            ) 0
        , Arr
            ( Span
                { start = AlexSourcePos
                    { line = 4
                    , col = 16
                    }
                , end = AlexSourcePos
                    { line = 4
                    , col = 19
                    }
                }
            ) 3
        , Obj
            ( Span
                { start = AlexSourcePos
                    { line = 4
                    , col = 19
                    }
                , end = AlexSourcePos
                    { line = 4
                    , col = 22
                    }
                }
            ) "ok"
        ]
    ]